core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -283,7 +283,7 @@ class BrokerServer(
         networkListeners.add(new Listener().
           setHost(if (Utils.isBlank(ep.host)) InetAddress.getLocalHost.getCanonicalHostName else ep.host).
           setName(ep.listenerName.value()).
-          setPort(socketServer.boundPort(ep.listenerName)).
+          setPort(ep.port).
```

Comment:
Thanks for the PR. Can you please add a test?

Reply:
Yes, sure. I will add a test.

Reply:
I added two tests to RaftClusterTest class. One for verifying that the configured advertised host and port are registered if the port is non zero. And another which in verifies that if the advertised port is 0, the bound port is used as actual advertised port. This is required to support binding the listener to any free port, which is used for example for the integration tests.